community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserterImpl.java

```diff
@@ -820,7 +795,7 @@ private boolean arrayContains( long[] ids, int cursor, int labelId )
     @Override
     public void createNode( long id, Map<String, Object> properties, Label... labels )
     {
-        if ( id < 0 || id > MAX_NODE_ID )
+        if ( id < 0 || id > recordFormats.node().getMaxId() )
```

Comment:
The RecordFormats methods weren't designed to be called in hot method like this one. For convenience those methods simply return a new RecordFormat instance every time. Rather figure out the max ID when constructing the BatchInserter and use that final value here instead.